### PR TITLE
snapcraft: depend on libffi for Linuxbrew

### PR DESCRIPTION
**Pull request by iMichka**

### CODE REVIEW

**Comment by iMichka** [comment]
> pkg_resources.DistributionNotFound: The 'python-apt' distribution was not found and is required by snapcraft

This will need a `python-apt` formula ...

File: Formula/snapcraft.rb [hunk 1]
```diff
@@ -18,6 +18,10 @@ class Snapcraft < Formula
   depends_on "libyaml"
   depends_on "lxc"
   depends_on :python3
+  # libffi is needed for cffi
+  depends_on "libffi" unless OS.mac?
+  # pkg-config helps "setup.py" find libffi
+  depends_on "pkg-config" => :build unless OS.mac?
```
Comment (sjackman) [inline Formula/snapcraft.rb]: Aesthetically, may I suggest:
```ruby
unless OS.mac?
  # pkg-config helps "setup.py" find libffi
  depends_on "pkg-config" => :build
  # libffi is needed for cffi
  depends_on "libffi"
end
```
Thanks for working on this!

**Comment by iMichka** [comment]
It's not the definitive version; the apt formula will need to be extracted to a separate PR first, but I want to see if the build works here first.

For apt, I installed just 2 folders with relevant c++ development files, but I have no clue if I should install more. Debian split all the files in different sub-packages, so I would need to read all their debian/*.install files, which I have no time for. I think if this works, it's a good starting point for the apt formula, and good enough to get merged.

**Comment by sjackman** [comment]
Why is `apt` required on Linux but not on macOS? Does it use a different package manager on macOS? If it uses `brew` on macOS, perhaps we could also use `brew` on Linux. All conjecture. I know nothing about Snap.

**Comment by sjackman** [comment]
And thanks for this work, Michka. A working replacement for `brew cask` on Linux would be awesome.

**Pull request closed without merging**